Sources/NIO/EventLoop.swift

```diff
@@ -50,6 +50,61 @@ public struct Scheduled<T> {
     }
 }
 
+public class RepeatedTask {
+    private let delay: TimeAmount
+    private var scheduled: Scheduled<EventLoopFuture<Void>>? {
```

Comment:
oh sorry, didn't notice that both of those are `var`. Did you run the tests with [TSan (thread sanitizer)](https://www.shinobicontrols.com/wp-content/uploads/2016/08/Enable_Sanitizer.png) enabled? I'm sure TSan will find the threading error here.

The assumption (and that should actually be documented) is that the user can `cancel` a `RepeatedTask` from any thread but this isn't the case here.